### DISCUSSION TITLE
Autocompletion only partially working

### DISCUSSION THREAD

**Discussion by boehm-e** [explosion/spaCy#7804] [Help: Other Questions] [answered]
## My Environment

- Operating System: Ubuntu 20.04
- Python Version Used: python 3.8.6
- spaCy Version Used: spacy 3.0.5
- Environment Information: VSCode 1.55.2

## The issue
Autocompletion is only partially working: the "nlp" object returned by spacy.load('xxx') "provides" autocompletion:
![Screenshot from 2021-04-15 15-36-25](https://user-images.githubusercontent.com/11290215/114878279-5f3db400-9e00-11eb-924e-541c1321be8f.png)

But none of the Doc, Token, Span imported spacy.tokens is able to produce autocompletion
![Screenshot from 2021-04-15 15-38-22](https://user-images.githubusercontent.com/11290215/114878537-a3c94f80-9e00-11eb-83ec-ca8ae3c920a1.png)

Is autocompletion for Doc/Token/Span available in some way?

PS: thank you for this amazing lib :heart:

**Comment by svlandeg** [comment]
Hi! As a maintenance note: this is a good question for the discussions forum, so I'll move it there. The original issue will be closed and will link through to the open discussion thread.

**Comment by svlandeg** [comment, marked as answer]
I think it may depend on whether or not your IDE can work with the Cython API. Because `Doc` etc are defined in `pyx` files rather than "normal" `.py` files.

With Pycharm (professional license), I do get the autocompletion:

![afbeelding](https://user-images.githubusercontent.com/8796347/114892311-ef81f600-9e0c-11eb-9c95-5bc3b2c04018.png)
